CatAnalyzer/plugins/TtbarSingleLeptonAnalyzer.cc

```diff
@@ -328,41 +402,73 @@ void TtbarSingleLeptonAnalyzer::analyze(const edm::Event& iEvent, const edm::Eve
   //---------------------------------------------------------------------------
 
   std::vector<cat::Muon> selectedMuons; 
+  std::vector<cat::Muon> vetoMuons; 
 
   Handle<edm::View<cat::Muon> > muons;
   iEvent.getByToken(muonToken_, muons); 
  
   for (unsigned int i = 0; i < muons->size() ; i++) { 
     const cat::Muon & muon = muons->at(i);
+
     if( IsTightMuon( muon) ) selectedMuons.push_back( muon);
+    else if( IsLooseMuon( muon) ) vetoMuons.push_back( muon); // does not Include selected muons
+
   }
 
   //---------------------------------------------------------------------------
-  //----------------------------------------------------------------
+  //---------------------------------------------------------------------------
   // Channel Selection
   //---------------------------------------------------------------------------
   //---------------------------------------------------------------------------
 
   TLorentzVector lepton;
-  int ch_tag  =999;
+  int ch_tag  = 999;
 
-  if(selectedMuons.size() == 1 && selectedElectrons.size() == 0){
+  if(selectedMuons.size()     == 1 && 
+     vetoMuons.size()         == 0 &&
+     selectedElectrons.size() == 0 && 
+     vetoElectrons.size()     == 0){
     lepton.SetPxPyPzE(selectedMuons[0].px(), selectedMuons[0].py(), selectedMuons[0].pz(), selectedMuons[0].energy());
     ch_tag = 0; //muon + jets
   }
 
-  if(selectedMuons.size() == 0 && selectedElectrons.size() == 1){
+  if(selectedMuons.size()     == 0 && 
+     vetoMuons.size()         == 0 &&
+     selectedElectrons.size() == 1 && 
+     vetoElectrons.size()     == 0){
     lepton.SetPxPyPzE(selectedElectrons[0].px(), selectedElectrons[0].py(), selectedElectrons[0].pz(), selectedElectrons[0].energy());
     ch_tag = 1; //electron + jets
   }
```

Comment:
This can be simplified,
```cpp
if ( vetoMuons.empty() && vetoElectrons.empty() && selectedMuons().size()+selectedElectrons().size() == 1 ) {
  if ( !selectedMuons.empty() ) {
    lepton.SetPxPyPzE(...);
    ch_tag = 0;
  }
  else if ( !selectedElectrons.empty() ) {
    lepton.SetPxPyPzE(...);
    ch_tag  = 1;
  }
}
```